packages/bot/src/struct/Command.ts

```diff
@@ -27,7 +38,110 @@ export interface Command<Type extends ApplicationCommandType = ApplicationComman
 	handle: (interaction: InteractionTypeMapping[Type]) => Awaitable<unknown>;
 }
 
-export type CommandConstructor = new (...args: any[]) => Command;
+export interface CommandWithSubcommands {
+	readonly interactionOptions: Omit<CommandBody<ApplicationCommandType.ChatInput>, 'options'>;
+	readonly subcommands: Map<string, Subcommand>;
```

Comment:
Quite frankly I'm not fond of this. I think we need some sort of boolean indicator to distinguish between `Command` and `CommandWithSubcommands`.

We should instead change `CommandHandler`'s `commands` to `Map<string, Command | CommandWithSubcommands | SubCommand>`

For `Command` and `CommandWithSubcommands` we just use `interactionOptions.name` to store them in the map, otherwise `${interactionOptions.name}-${subcommand.interactionOptions.name}`.

When `handleCommand` is called, if we find a `Command` we proceed normally by calling its `handle`.

If we find a `CommandWithSubcommands`, we `this.commands.get(${command.name}-${interaction.getFocused()})` (pseudo-code, I forget how we get which subcommand is selected), and use its `handle`.


For autocomplete, for `Command` it works as always. For `CommandWithSubcommands`, it's very similar to what I described above, resolve the `Subcommand` and try doing `subcommand.handleAutocomplete.?(...args) ?? command.handleAutoComplete.?(...args)` - where `command` is a `CommandWithSubcommands`, of course.

Reply:
I implemented the changes you requested but it may not be in the way that you've requested exactly cause I may have misinterpreted or assumed certain things that may not have been explicitly explained. For example:
> Quite frankly I'm not fond of this. I think we need some sort of boolean indicator to distinguish between `Command` and `CommandWithSubcommands`.
I may have taken the boolean indicator too literally but implemented it how I thought through it anyway just to have some sort of basis for differentiating commands until explained with a better implementation idea.